docs/packages.md

```diff
@@ -11,6 +11,22 @@ All LinuxKit packages are:
 - Derived from well-known (and signed) sources for repeatable builds.
 - Build with multi-stage builds to minimise their size.
 
+## Package source
+
+A package source consists of a directory containing two files:
+
+- `build.yml`: contains metadata associated with the package
+- `Dockerfile`: contains the steps to build the package.
+
+`build.yml` contains the following fields:
+
+- `image` _(string)_: *(mandatory)* The name of the image to build
+- `org` _(string)_: The hub/registry organisation to which this package belongs
+- `arches` _(list of string)_: The architectures which this package should be built for (valid entries are `GOARCH` names)
+- `gitrepo` _(string)_: The git repository where the package source is kept.
+- `network` _(bool)_: Allow network access during the package build (default: no)
+- `trust` _(bool)_: Enable Docker content trust for this package (default: yes)
```

Comment:
its weird having booleans that default to `true` - should this be `notrust` and `nocache`?

Reply:
also means cli flags can just be `--notrust` not `-trust=0`

Reply:
or `disable-` rather than `no`

Reply:
I don't think it's at all weird, IMHO inverting the sense of the name is odd and leads to double negatives in the code `if !foo.notrust` which just make it harder to follow.

Reply:
`disable` has similar problems to `no`. The `--notrust` things is IMHO just a short-coming of the golang `flag` package. I've been considering looking for a better one since it also doesn't let you know if something was set on the command line or not leading to hoop jumping.

Reply:
The CLI needs to be able to overload the yml, which could have an explicit `true`, so it's not as simple as inverting the option only. I could add both `--trust` and `--notrust` and handle reconciling when both are used (since `flagset.Visit` goes in cli order it should be reasonably easy to implement last-one-wins semantics).

Reply:
I still prefer boolean defaults to always be false. Having both cli flags is ok.

Reply:
I dont think the code argument is any good, you need both cases anyway so dont invert the sense.

Reply:
If defaults randomly vary between true and false I have to either be really verbose and write everything out or RTFM every time I make a config file. If they are always false its obvious.

Reply:
I'll try it and see how it looks

Reply:
I think `-disable-content-trust` for the CLI and `disable-content-trust` for the YAML option is better. It's what `docker` and `moby` is using and it's sufficiently lengthy and awkward to type for someone to really want to disable it. For this option it doesn't seem to make sense, where someone would disable content trust in the YAML and the you want to override it in the CLI to enable it. That seems like an odd use case.

Reply:
Like that name, I think I might as well support the odd override since it's not impossible, just unusual and I'll have the pattern from the other bools anyway. I could imagine someone not wanting to routinely sign stuff but only specific builds which they decide are good by some metric.

Reply:
agree with `disable-content-trust` consistency, and it makes the default boolean case much cleaner